ast/term.go

```diff
@@ -2392,6 +2407,59 @@ func filterObject(o Value, filter Value) (Value, error) {
 	}
 }
 
+// NOTE(philipc): The only way to get an ObjectKeyIterator should be
+// from an Object. This ensures that the iterator can have implementation-
+// specific details internally, with no contracts except to the very
+// limited interface.
+type ObjectKeysIterator interface {
```

Comment:
💭 Do we even need to make the type public? But I guess it's OK.

Reply:
I'm not actually sure that would happen with a non-public interface. Not worth the hassle, I think.